system/View/Parser.php

```diff
@@ -572,12 +569,18 @@ public function setDelimiters($leftDelimiter = '{', $rightDelimiter = '}'): Rend
 	 */
 	protected function replaceSingle($pattern, $content, $template, bool $escape = false): string
 	{
-		// Any dollar signs in the pattern will be mis-interpreted, so slash them
+		// Any dollar signs in the pattern will be misinterpreted, so slash them
 		$pattern = addcslashes($pattern, '$');
 
+		// Flesh out the main pattern from the delimiters and escape the hash
+		if (preg_match('/^(#)(.*)(#(m?s)?)$/', $pattern, $parts))
```

Comment:
I think it need note with `@see https://regex101.com/path` to show the regex demo, eg: 

```php
// @see https://regex101.com/r/gV1eVk/1
```

so in the future, when there is a bug fix on regex, it will be easier to check/update.